### ISSUE TITLE
False Positive for CS1674 error in a case of void Dispose(); method.

### ISSUE THREAD

**Issue by dmitriyse** [dotnet/roslyn#40353]
**Version Used**: 
Visual Studio 2019 16.4.1 or Visual Studio 2019 16.5.0 Preview 1.0
**Steps to Reproduce**:

1. Create .Net Core Console Project (netcoreapp3.1)
2. Edit Program.cs:
```CSharp
using System;
using System.Threading.Tasks;

namespace PatternBasedUsingTest
{
    public class Foo
    {
        public void Dispose(){}
    }

    public class FooAsync
    {
        public async ValueTask DisposeAsync() { }
    }

    class Program
    {
        static void Main(string[] args)
        {
            using (var f = new Foo()) // Gives CS1674 error
            {

            }
            Console.WriteLine("Hello World!");
        }

        static async ValueTask Test()
        {
            await using (var fa = new FooAsync()) // Works !!!
            {

            }
        }
    }
}
```
3. Compile

**Expected Behavior**:
Compiled successfully
**Actual Behavior**:
Error CS1674	'Foo': type used in a using statement must be implicitly convertible to 'System.IDisposable' or implement a suitable 'Dispose' method.

**Comment by ViIvanov** [comment]
Looks like this is by design: [Enhanced Using](https://github.com/dotnet/roslyn/blob/master/docs/features/enhanced-using.md):
> [Pattern-based asynchronous disposal](https://github.com/dotnet/roslyn/blob/master/docs/features/enhanced-using.md#Pattern-based-asynchronous-disposal) allows a type to opt into asynchronous disposal without needing to implement the `IAsyncDisposable` interface.
> [Pattern-based disposal for ref structs](https://github.com/dotnet/roslyn/blob/master/docs/features/enhanced-using.md#Pattern-based-disposal-for-ref-structs) allows `ref structs` to opt into disposal without needing to implement the `IDisposable` interface.

So, pattern-based `IDisposable` should works only on `res struct`s and pattern-based `IAsyncDisposable` on any type.

**Comment by jaredpar** [comment]
Correct this is by design.
